### DISCUSSION TITLE
Anybody running an Intel arc 750 or 770?

### DISCUSSION THREAD

**Discussion by RahulVivekNair** [ggml-org/llama.cpp#1923] [General]
If anyone is running these cards for their vram capacity , what is your experience like? How many iterations/ms are you getting through opencl offloading? Does it work with UI's like oobabooga and is it worth getting one?

**Comment by KerfuffleV2** [comment]
This is gaming oriented, but it seems like it's about half the speed of a 3090: https://gpu.userbenchmark.com/Compare/Nvidia-RTX-3090-vs-Intel-Arc-A770/4081vsm1850973

It also has 33% less memory. However, it costs about half as much. Assuming the OpenCL performance is in line with the gaming performance, it could possibly make sense to get two of them and use stuff like GGML GPU splitting feature. However, the cards have 250 watt TDP so that's a huge amount of power.

The cheapest one I found was $339, used 3090s are $700-800.

I didn't find anything much about OpenCL usage, but it seems like the card is pretty well supported on Linux.

**Reply by Green-Sky** [reply]
looks like Intel MKL (BLAS) is supported. So at least prompt ingestion should be good.

**Reply by JohannesGaessler** [reply]
>get two of them and use stuff like GGML GPU splitting feature.

As far as I know OpenCL currently does not support multi GPU.

**Comment by ianscrivener** [comment]
I checked out the the specs on these recently. Yes, price for performance is very good. Though am I right to assume they are using 50-100% more electricity than an equaivalent nvidia card? I'm SUPER impressed with the performance versus energy of my MacOS M2. The Amazon Inferentia etc ARM chips are also super energy efficient.

**Comment by taka-tuos** [comment]
I have an A380 (ASRock Challenger) and tried llama.cpp. 7B (vicuna-1.0, Q4_0) produced a terrible result of 240ms/tok. (ArchLinux, E5-2670 v3 with DDR4-2133 32GB)
P104-100 hits 30ms/tok in the same environment. (Slightly unfair comparison: A380 cost 22000 Yen brand new, while P104-100 costs only 3000 Yen because I co-bought a mining rig and took it apart lol)

**Reply by netrunnereve** [reply]
I wonder if this is due to the lack of Resizable BAR support on your v3 Xeon.

**Reply by taka-tuos** [reply]
Maybe so, I'll try the ReBAR BIOS Mod tonight.
Currently TFlops/tok/s is 3x worse than P104-100 on CLBlast (roughly 0.3 vs 1.0, P104 is about 0.2 for cuBLAS). However, since the entire model is in VRAM, I doubt ReBAR alone will give a 3x performance jump...

**Reply by taka-tuos** [reply]
ReBAR Enabled. 240ms/tok -> 200ms/tok

**Reply by taka-tuos** [reply]
P104 on CLBlast was also remeasured and it was 100ms/tok (0.67TFlops/tok/s). The A380's 200ms/tok is 0.92TFlops/tok/s, so considering the driver's maturity and VRAM bandwidth, it may not be that bad?

**Reply by tangjinchuan** [reply]
Currently, clBLAST 1.6.1 is not tuned for any Intel ARC GPUs which could cause performance-related problems. Recently, the project has received ARC 770's tunned results from the community. If you are interested in tunning your device to possibly speed up your tokens, please see   https://github.com/ggerganov/llama.cpp/discussions/1688

**Reply by taka-tuos** [reply]
Thanks for the useful information, I'll try tuning CLBlast on my A380 this weekend.

**Reply by DarwinAnim8or** [reply]
@taka-tuos did you ever re-try running llama on the a380? curious if it's doable

**Reply by taka-tuos** [reply]
I'll be trying out SYCL in the next few weeks.
As far as I know, the only 75W, 6GB (that can run 8b Q3_K_M) is the A380, so if it works, I think it's a pretty attractive option.

**Reply by DarwinAnim8or** [reply]
> I'll be trying out SYCL in the next few weeks. As far as I know, the only 75W, 6GB (that can run 8b Q3_K_M) is the A380, so if it works, I think it's a pretty attractive option.

I have an A380 now, and tried to run SYCL on Ubuntu but it spat an error at the end :(

**Reply by DarwinAnim8or** [reply]
Update-- seems that it's largely dependent on what model you're loading and at what Quant, Q4_K_M works, but Q5_K_S didn't play nice, for example.

**Comment by tarunmcom** [comment]
Run LLama-2 13B, very fast, Locally on Low Cost Intel's ARC GPU , iGPU and on CPU:- https://youtu.be/FRWy7rzOsRs

**Reply by taka-tuos** [reply]
is this your youtube channel? Could you please use 7B-q4_0 and set n_predict to 64 and check tok/ms?

**Reply by delock** [reply]
Hi @tarunmcom  from your video I saw you are using A770M and the speed for 13B is quite decent..  I have tuned for A770M in CLBlast but the result runs extermly slow.  Also when I try to copy A770 tuning result, the speed to inference llama2 7b model with q5_M is not very high (around 5 tokens/s), which is even slower than using 6 Intel 12gen CPU P cores.

Are you using A770 tunning result in CLBlast or do you tune for A770M by yourself?  would like to know whether the tunning result can be shared.

**Reply by NeoZhangJianyu** [reply]
@delock 
You could try with llama.cpp for SYCL backend, refer to https://github.com/ggerganov/llama.cpp/blob/master/docs/backend/SYCL.md
It's faster than CLBlast on Intel GPU.

**Reply by HumerousGorgon** [reply]
Using the sycl backend, whenever it tries to allocate >4GB of VRAM to my A770, the model fails loading. Trying to load anything higher than Q4 means a failure, even though I have the available VRAM for it. Any idea on how to solve this?

**Reply by NeoZhangJianyu** [reply]
Please share the whole log, including the cmd line.
Please report an issue, title with [SYCL].
I will monitor the issue in general.

**Reply by HumerousGorgon** [reply]
@NeoZhangJianyu
Just posted my error with relevant logs. thank you.

**Comment by tarunmcom** [comment]
here is video how to run using sycl its father than CLBLAST :-
https://www.youtube.com/watch?v=Q7t4CmziaqA
